### PR TITLE
Added support for using queries in React (useQuery). Also refactored Todo example to use new code.

### PR DESCRIPTION
**Pull request by Martinsos**

### CODE REVIEW

File: waspc/data/Generator/templates/react-app/src/queries/_query.js [hunk 1]
```diff
@@ -18,4 +18,6 @@ const {= queryFnName =} = async ({ args, context }) => {
   }
 }
 
+{= queryFnName =}.useQueryKey = '{= queryRoute =}'
```
Comment (matijaSos) [inline waspc/data/Generator/templates/react-app/src/queries/_query.js:21]: aha ok this is sth required by 'useQuery' I suppose.

File: waspc/data/Generator/templates/react-app/src/queries/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useQuery as rqUseQuery } from 'react-query'
+
+export const useQuery = (queryFn, queryFnArgs, config) => {
+  if (queryFn === 'undefined') {
```
Comment (matijaSos) [inline waspc/data/Generator/templates/react-app/src/queries/index.js:4]: did we intend here to compare with a string, and not just `undefined`?

File: waspc/data/Generator/templates/react-app/src/queries/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    throw new Error('useQuery requires queryFn to be specified.')
+  }
+  if (!queryFn.useQueryKey) {
+    throw new Error('queryFn needs to have useQueryKey property defined.')
+  }
+
+  const rqResult = rqUseQuery({
+    queryKey: [queryFn.useQueryKey, queryFnArgs],
+    queryFn: (_key, args) => queryFn(args),
```
Comment (matijaSos) [inline waspc/data/Generator/templates/react-app/src/queries/index.js:13]: `key` is not used, that is why it is underscored?

File: waspc/examples/todoApp/ext/Todo.js [hunk 4]
```diff
@@ -1,68 +1,92 @@
...
+    const task = { isDone: false, description }
+    await createTask(task)
+    refetch()
+  }
+
+  const handleNewTaskSubmit = async (event) => {
+    event.preventDefault()
+    try {
+      await createNewTask(newTaskDescription)
+      setNewTaskDescription('')
```
Comment (matijaSos) [inline waspc/examples/todoApp/ext/Todo.js:26]: `''` could be extracted into some variable as e.g. `taskDefaultDescription` or sth like that, now it is being duplicated (here and in `useState` in line 8)

File: waspc/examples/todoApp/ext/state.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+const state = {}
```
Comment (matijaSos) [inline waspc/examples/todoApp/ext/state.js:1]: aha is this global local state? Is that on server or on the client? Maybe write that in the comments.

**Review by matijaSos — APPROVED** [review]
cool, minor stuff, looking good!

File: waspc/data/Generator/templates/react-app/src/queries/_query.js (hunk 1 above)
Comment (Martinsos) [reply waspc/data/Generator/templates/react-app/src/queries/_query.js:21]: It requires the key, and we want it to be unique per query. In our case, route is unique, so we are using that.

File: waspc/data/Generator/templates/react-app/src/queries/index.js (hunk 3 above)
Comment (Martinsos) [reply waspc/data/Generator/templates/react-app/src/queries/index.js:13]: Yes! I was inspired by Haskell here.

File: waspc/examples/todoApp/ext/Todo.js (hunk 4 above)
Comment (Martinsos) [reply waspc/examples/todoApp/ext/Todo.js:26]: Very good point! I named it defaultNewTaskDescription.

**Pull request merged**